components/utils/fetchAllArticles.jsx

```diff
@@ -0,0 +1,18 @@
+import axios from 'axios'
+
+export default async function fetchAllArticles ({setArticles, queryString, sortDirection}) {
+    if (queryString){
```

Comment:
Nice that you've made this util function flexible to sometimes accept parameters
You could make your code a bit more succinct here using the axios `params` functionality
https://notes.northcoders.com/courses/js-front-end/axios#axios-params